### PR TITLE
ICU-22314 Refactor GH CI jobs into workflows triggered by modified paths

### PR DESCRIPTION
**Pull request by echeran**
Testing PRs in personal fork:

- [docs only change](https://github.com/echeran/icu/pull/55)
- [ICU4C only change](https://github.com/echeran/icu/pull/56)
- [ICU4J only change](https://github.com/echeran/icu/pull/57)

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22314
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by mihnita — APPROVED** [review]
LGTM, Thank you!

**Comment by mihnita** [comment]
Hmm, I wonder if there is something to cover the "transition" between icu4c and icu4j.
When we generate binary resources (.res & co) from C to Java, there are new files in icu4j, so that's OK.
But do we test the stuff in /tools (the root one, not icu4j/tools)
It looks like we don't test that now (before this PR)

(the LGTM / approve still stands, as this is something we don't do today)

File: .github/workflows/icu_common.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          java-version: '8'
+      # Download all of the artifacts needed for the code and build plugins, but
+      # exclude any needed by profiles depending on system artifacts
+      - name: Download all artifacts
+        run: |
+          cd icu4j;
+          mvn ${SHARED_MVN_ARGS} dependency:go-offline -P '!old_jdk_taglet'
+
+  # Verify icu4c release tools buildability.
+  icu4c-release-tools:
```
Comment (markusicu) [inline .github/workflows/icu_common.yml:71]: are these really "ICU4C release tools"?

File: .github/workflows/icu_docs.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          generator_config_file: docs/_config.yml
+      - name: Build with Jekyll
+        # Outputs to the './_site' directory by default
+        run: |
+          cd docs  # root directory of markdown, also contains Jekyll configs, etc.
+          bundle install
+          # The baseurl arg is parsed from the `baseurl` field of _config.yml.
+          bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path }}"
+        env:
+          JEKYLL_ENV: production
```
Comment (markusicu) [inline .github/workflows/icu_docs.yml:54]: add a line feed?

**Review by markusicu — APPROVED** [review]
lgtm except for two nits/questions

File: .github/workflows/icu_common.yml (hunk 1 above)
Comment (markusicu) [reply .github/workflows/icu_common.yml:71]: We looked, this is indeed for the *ICU4C* API change report:
- https://github.com/unicode-org/icu/blob/main/tools/release/java/pom.xml
- https://github.com/unicode-org/icu/blob/main/tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java

So the name seems fine.
Or we could shorten it to "release tools".

File: .github/workflows/icu_docs.yml (hunk 2 above)
Comment (echeran) [reply .github/workflows/icu_docs.yml]: Done, I think.

**Review by markusicu — APPROVED** [review]

**Pull request merged**